pkg/api/v1alpha1/work_types.go

```diff
@@ -0,0 +1,156 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// WorkSpec defines the desired state of Work
+type WorkSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Workload represents the manifest workload to be deployed on spoke cluster
+	Workload WorkloadTemplate `json:"workload,omitempty"`
+}
+
+// WorkloadTemplate represents the manifest workload to be deployed on spoke cluster
+type WorkloadTemplate struct {
+	// Manifests represents a list of kuberenetes resources to be deployed on the spoke cluster.
+	// +optional
+	Manifests []runtime.RawExtension `json:"manifests,omitempty" protobuf:"bytes,2,rep,name=manifests"`
+}
+
+// WorkStatus defines the observed state of Work
+type WorkStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Conditions contains the different condition statuses for this work.
+	// Valid condition types are:
+	// 1. Applied represents workload in Work is applied successfully on spoke cluster.
+	// 2. Progressing represents workload in Work is being applied on spoke cluster.
+	// 3. Available represents workload in Work exists on the spoke cluster.
+	// 4. Degraded represents the current state of workload does not match the desired
+	// state for a certain period.
+	Conditions []StatusCondition `json:"conditions"`
+
+	// ManifestConditions represents the conditions of each resource in work deployed on
+	// spoke cluster.
+	// +optional
+	ManifestConditions []ManifestCondition `json:"manifestConditions,omitempty"`
+}
+
+// GroupVersionKind is the same as schema.GroupVersionKind but with json tag
+type GroupVersionKind struct {
+	// Group is the group of the resource.
+	Group string `json:"group,omitempty"`
+
+	// Version is the version of the resource.
+	Version string `json:"version,omitempty"`
+
+	// Kind is the kind of the resource.
+	Kind string `json:"kind,omitempty"`
+}
+
+// NamespacedName is the same as typed.NamespacedName but with json tag
+type NamespacedName struct {
+	// Namespace is the namespace of the resource
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name is the name of the resource
+	Name string `json:"name,omitempty"`
+}
+
+// ResourceIdentifier provides the identifiers needed to interact with any arbitrary object.
+type ResourceIdentifier struct {
+	// Ordinal represents an index in manifests list
```

Comment:
you'll want to explain why in the comments.  This exists in case you cannot successfully parse those manifests and need a way to refer to "the one that's busted"

Reply:
added some comments